src/workerd/io/compatibility-date.c++

```diff
@@ -196,12 +200,11 @@ void compileCompatibilityFlags(kj::StringPtr compatDate, capnp::List<capnp::Text
       // it redundant, because at a future date it won't be redundant, and someone could want to
       // set the flag early to make sure they don't forget later.
     }
-    if (dateValidation == CompatibilityDateValidation::CODE_VERSION &&
-        enableByFlag && !enableByDate && enableDate == nullptr) {
+    if (dateValidation == CompatibilityDateValidation::CURRENT_DATE_FOR_CLOUDFLARE &&
+        (isExperimental && !allowExperimentalFeatures)) {
       errorReporter.addError(kj::str(
-          "The compatibility flag ", enableFlagName, " is experimental and may break or be "
-          "removed in a future version of workerd. To use this flag, you must pass --experimental "
-          "on the command line."));
+          "The compatibility flag ", enableFlagName, " is experimental and cannot yet be used in ",
```

Comment:
For `workerd` (`CODE_VERSION` case), let's please keep the error message the same as it was before. Only use this new error text in the Cloudflare case (`CURRENT_DATE_FOR_CLOUDFLARE`).

Reply:
Fixed